### PR TITLE
[toolchain] introduce OT_TOOL_WEAK definition

### PR DESCRIPTION
**Pull request by LuDuda**

### CODE REVIEW

File: src/core/api/tasklet_api.cpp [hunk 1]
```diff
@@ -52,3 +52,7 @@ bool otTaskletsArePending(otInstance *aInstance)
 {
     return aInstance->mIp6.mTaskletScheduler.AreTaskletsPending();
 }
+
+OT_TOOL_WEAK void otTaskletsSignalPending(otInstance *)
+{
+}
```
Comment (jwhui) [inline src/core/api/tasklet_api.cpp]: nit: add newline

File: src/core/api/tasklet_api.cpp [hunk 2]
```diff
@@ -52,3 +52,10 @@ bool otTaskletsArePending(otInstance *aInstance)
 {
     return aInstance->mIp6.mTaskletScheduler.AreTaskletsPending();
 }
+
+#ifndef _MSC_VER
```
Comment (LuDuda) [inline src/core/api/tasklet_api.cpp]: Do you have any better idea on how to implement weak functions in MSVS? Appveyor seems to not like `__declspec(selectany)` attribute. @nibanks ?

Comment (nibanks) [inline src/core/api/tasklet_api.cpp]: I don't know. Likely not supported in Windows kernel mode.

Comment (LuDuda) [inline src/core/api/tasklet_api.cpp]: @jwhui Can we then accept this fact and use this ifdefs, or do you see some more clever way - like creating a new .cpp file with weak functions that is not built by MSVS?

Comment (jwhui) [inline src/core/api/tasklet_api.cpp]: Works for me.

**Pull request merged**